lib/node_modules/@stdlib/blas/base/dscal/src/dscal_cblas.c

```diff
@@ -31,3 +32,17 @@
 void API_SUFFIX(c_dscal)( const CBLAS_INT N, const double alpha, double *X, const CBLAS_INT stride ) {
 	API_SUFFIX(cblas_dscal)( N, alpha, X, stride );
 }
+
+/**
+* Multiplies a double-precision floating-point vector `X` by a constant using alternative indexing semantics.
+*
+* @param N       number of indexed elements
+* @param alpha   scalar
+* @param X       input array
+* @param stride  index increment
+* @param offset  starting index
+*/
+void API_SUFFIX(c_dscal_ndarray)( const CBLAS_INT N, const double alpha, double *X, const CBLAS_INT stride, const CBLAS_INT offset ) {
+	X += stdlib_strided_min_view_buffer_index( N, stride, offset ); // adjust array pointer
```

Comment:
@aman-095 Before L47, we should convert `stride` to a positive stride before calling into CBLAS, which may or may not support negative strides. By always passing in a positive stride, we avoid backward compat breaks.